src/sagemaker_training/process.py

```diff
@@ -285,6 +287,9 @@ def check_error(cmd, error_classes, processes_per_host, cwd=None, capture_error=
         stderr = " ".join(output)
         # remove duplicate while preserve order
         stderr = "\n".join(list(dict.fromkeys(stderr.split("\n")))).strip()
+        if any(path in traceback.format_exc() for path in SM_TRAINING_COMPILER_PATHS):
```

Comment:
This logic seems flawed. 

1. Here we only have access to the return code, stdout/err of the forked training process. At this point forked training process has already finished and its stack has been cleared. Try using the stderr directly instead of the stack trace. The stack will already be dumped to stderr.

2. This logic should come into play only for failed processes. The correct place for this would be between line 305-326 along with the string based failure attribution. You should design how string-based FA should work alongside stack-based FA.

Reply:
changed it to add the message SMTrainingCompiler Error only during the failed process. Need to understand the string based failure attribution. Will discuss and add a new PR. Resolving this now.